metrics/core/src/main/java/tech/pegasys/pantheon/metrics/RunnableCounter.java

```diff
@@ -48,9 +49,14 @@ public void inc() {
   public void inc(final long amount) {
     stepCounter.addAndGet((int) amount);
     backedCounter.inc(amount);
+    total++;
```

Comment:
This isn't thread safe so total will wind up being the wrong value. 

Actually `stepCounter` isn't thread safe either.  It should be a `final AtomicLong` then this method becomes something like:
```
backedCounter.inc(amount);
if (stepCounter.addAndGet(amount) % step == 0) {
  task.run();
}
```

And then you can just get the total with `stepCounter`.

Reply:
Yes it is true, this is what i did for the first implementation of this class. But during the review someone said that if we don't need the total then no need to use a modulo just store the incremental counter and reset to zero when hit the step. But now we need the total so i will implement this approach.

Reply:
Right, it's not that we need the total - it's that it's not thread-safe to assign the new `AtomicInteger` (nor is it safe to downcast the long to an int when incrementing it).  You could wind up with all kinds of weird exceptions coming out of it because the `AtomicInteger` could be exposed to other threads before its constructor has actually completed (compilers and CPUs are free to reorder instructions as there's no memory fences involved).

Reply:
Done.